src/mol-io/writer/mol2/encoder.ts

```diff
@@ -62,146 +61,224 @@ export class Mol2Encoder extends LigandEncoder {
         StringBuilder.writeSafe(this.out, `@<TRIPOS>SUBSTRUCTURE\n1 ${name} 1\n`);
     }
 
-    // private toArray(map: Map<string, { order: number, flags: number }>): BondData[] {
-    //     const array: BondData[] = [];
-    //     map.forEach((v, k) => array.push({ label_atom_id: k, order: v.order, aromatic: BondType.is(BondType.Flag.Aromatic, v.flags) }));
-    //     return array;
-    // }
+    private count<K, V>(map: Map<K, V>, predicate: (k: K, v: V) => boolean): number {
```

Comment:
Rewriting it to 
```ts
private count<K, V, C>(map: Map<K, V>,  ctx: C, predicate: (k: K, v: V, ctx: C) => boolean): number 
```
would help a lot to avoid creating some closures. Generally, you want to avoid creating closures as much as possible in inner loops.

For example, ``this.count(adjacentBonds, (k, _v) => this.isNonMetalBond(k))`` would become ``this.count(adjacentBonds, this, (k, _v, ctx) => ctx.isNonMetalBond(k))``.

Reply:
Thanks for the direction. I'll read up some on the matter.